simplekv/net/_azurestore_new.py

```diff
@@ -95,7 +96,14 @@ def _get(self, key):
             return downloader.readall()
 
     def _has_key(self, key):
-        return self.blob_container_client.exists(key)
+        # https://github.com/Azure/azure-sdk-for-python/issues/9507
+        blob_client = self.blob_container_client.get_blob_client(key)
+        try:
```

Comment:
To mimic existing similar functions, I'd rather propose to implement it this way:
```python
with map_azure_exceptions(key, ("BlobNotFound",)):
    blob_client.get_blob_properties()
    return True
return False
```